### PR TITLE
chore: improve performance of ClearInternal

### PR DESCRIPTION
**Pull request by romange**
Before:
`BM_Clear/elements:32000     418763 ns       418749 ns         9965`

After:
`BM_Clear/elements:32000     323252 ns       323239 ns        12790`

### CODE REVIEW

File: src/core/dense_set.cc [hunk 1]
```diff
@@ -255,6 +268,39 @@ void DenseSet::CloneBatch(unsigned len, CloneItem* items, DenseSet* other) const
...
+        continue;
+
+      if (src.ptr.IsObject()) {
+        ObjDelete(src.ptr.Raw(), src.has_ttl);
+      } else {
+        auto& dest = items[dest_id++];
+        DenseLinkKey* link = src.ptr.AsLink();
+        dest.obj = link->Raw();
+        dest.has_ttl = link->HasTtl();
+        dest.ptr = link->next;
```
Comment (kostasrim) [inline src/core/dense_set.cc:293]: Same here second time I see this maybe worth making it a function. Actually we have the exact same boilerplate in the body of the for-loop from line 277 to 297. See https://github.com/dragonflydb/dragonfly/pull/3862/files#diff-5685b15a1d75a2c80b43908203ee74ab673f9a748f1dce35ad1f018bb656eb6aR236 with the only difference the function that gets called `clone` vs `ObjDelete`

File: src/core/dense_set.cc [hunk 2]
```diff
@@ -180,21 +180,34 @@ void* DenseSet::PopDataFront(DenseSet::ChainVectorIterator it) {
...
+    dest.has_ttl = ptr.HasTtl();
+
+    PREFETCH_READ(ptr.Raw());
+    if (ptr.IsObject()) {
+      dest.obj = ptr.Raw();
+      dest.ptr.Reset();
+    } else {
+      dest.ptr = ptr;
+      dest.obj = nullptr;
+    }
```
Comment (kostasrim) [inline src/core/dense_set.cc:202]: You merge very similar code in https://github.com/dragonflydb/dragonfly/pull/3862/files#diff-5685b15a1d75a2c80b43908203ee74ab673f9a748f1dce35ad1f018bb656eb6aR342 maybe worth extracting it into a function with an optional callback ?

File: src/core/dense_set.cc [hunk 3]
```diff
@@ -180,21 +180,34 @@ void* DenseSet::PopDataFront(DenseSet::ChainVectorIterator it) {
...
+    if (len == kArrLen) {
+      ClearBatch(kArrLen, arr);
+      len = 0;
     }
```
Comment (kostasrim) [inline src/core/dense_set.cc:206]: Nice batching here :) Just curious why `constexpr unsigned kArrLen = 32;` ? Have you played around and increased this batch limt ?

**Review by kostasrim — APPROVED** [review]
LGTM some minor comments in regards to code duplication

File: src/core/dense_set.cc (hunk 1 above)
Comment (romange) [reply src/core/dense_set.cc:293]: it's not the same code because here we also call `FreeLink`

File: src/core/dense_set.cc (hunk 3 above)
Comment (romange) [reply src/core/dense_set.cc:206]: i did yeah. it can probably be differrent for differrent CPUs but it more or less near a sweet point

**Pull request merged**